### PR TITLE
Refactor base modules and recipies to facilitiate moving feature to separate libraries

### LINKED ISSUE #10635: Refactor base modules and recipies to facilitiate moving feature to separate libraries
In order to make our transtioning to feature driven architecture in 3.x, we need to decouple and change base core and UI modules of Spartacus.

We also would like to achieve reference structure that will be ready for next major release.

### PR DESCRIPTION
**Pull request by dunqan**
In order to make our transtioning to feature driven architecture in 3.x, we need to decouple and change base core and UI modules of Spartacus.

We also would like to achieve reference structure that will be ready for next major release.

Closes #10635

### CODE REVIEW

File: projects/storefrontlib/src/base-ui.module.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import { SeoModule } from './cms-structure/seo/seo.module';
+import { LayoutModule } from './layout/layout.module';
+import { SkipLinkModule } from './layout/a11y/skip-link/skip-link.module';
+import { KeyboardFocusModule } from './layout/a11y/keyboard-focus/keyboard-focus.module';
+import { MediaModule } from './shared/components/media/media.module';
+
+@NgModule({
+  declarations: [],
+  imports: [
+    CommonModule,
```
Comment (marlass) [inline projects/storefrontlib/src/base-ui.module.ts]:
```suggestion
```

File: projects/core/src/base-core.module.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    ConfigModule.forRoot(),
+    ConfigInitializerModule.forRoot(),
+    ConfigValidatorModule.forRoot(),
+    I18nModule.forRoot(),
+    CmsModule.forRoot(),
+    GlobalMessageModule.forRoot(),
+    ProcessModule.forRoot(),
+    FeaturesConfigModule.forRoot(),
+    SiteContextModule.forRoot(), // should be imported after RouterModule.forRoot, because it overwrites UrlSerializer
+    OccModule.forRoot(),
```
Comment (marlass) [inline projects/core/src/base-core.module.ts]: It looks like we will remove things from the `OccModule` with next libraries and making it breaking changes. Would move it to spartacus-setup

**Review by marlass — CHANGES_REQUESTED** [review]

File: projects/storefrontlib/src/base-ui.module.ts (hunk 1 above)
Comment (dunqan) [reply projects/storefrontlib/src/base-ui.module.ts]: thx 👍

File: projects/core/src/base-core.module.ts (hunk 2 above)
Comment (dunqan) [reply projects/core/src/base-core.module.ts]: fixed 👍

**Review by marlass — DISMISSED** [review]
Just adjust the deprecation version and add link to page describing new structure

**Review by marlass — DISMISSED** [review]

**Review by marlass — APPROVED** [review]

**Pull request merged**